### DISCUSSION TITLE
get Excel sheet names  and combine sheets  by function  st_read

### DISCUSSION THREAD

**Discussion by JohnBianQi** [duckdb/duckdb#11352] [Ideas]
Hi,
Is there any way that we could get all sheetNames when using st_read function  import excel files?
Say, there's a excel workbook with 3 worksheets,  all 3 worksheets have datas and have the same schema.
I want  to get the sheet names, column names and combine all 3 worksheets in one.
DuckDB has any solution to make it? 
Or st_read has the same function "union by name" like it in read_csv?

**Comment by DeoLeung** [comment]
similar request

when reading an excel, we might not know all the sheet names

1. it will be great to have options to read in all sheets (may with different schemas) at once
2. function to get all the sheet names and read in on demand, it may result in round trip of reading file if over httpfs :)

**Comment by abubelinha** [comment]
I agree with having a way to getting all sheets, but importing them as separate duckdb tables (just like they are in Excel).
I see a possible problem here: which names should these new tables get? (as it could happen that duckdb database already has one or more tables with the same name)  
Perhaps there should be an optional prefix string parameter, so duckdb prepends that string to new created tables' names.

Merging all tables into one is a more particular use case, so I would make that an option (i.e. another boolean parameter or so)

**Comment by themichaelusa** [comment]
Bumping this. Would be extremely useful if possible.
